src/table.ts

```diff
@@ -1504,23 +1538,44 @@ Please use the format 'prezzy' or '${instance.name}/tables/prezzy'.`);
     );
     const mutationErrorsByEntryIndex = new Map();
 
-    const onBatchResponse = (
-      err: ServiceError | PartialFailureError | null
-    ) => {
-      // TODO: enable retries when the entire RPC fails
-      if (err) {
-        // The error happened before a request was even made, don't retry.
-        callback(err);
-        return;
+    const isRetryable = (err: ServiceError | null) => {
+      // Don't retry if there are no more entries or retry attempts,
+      // or the error happened before a request was made.
+      if (
+        pendingEntryIndices.size === 0 ||
+        numRequestsMade >= maxRetries ||
+        numRequestsMade === 0
+      ) {
+        return false;
       }
-      if (pendingEntryIndices.size !== 0 && numRequestsMade <= maxRetries) {
-        makeNextBatchRequest();
+      return !err || IDEMPOTENT_RETRYABLE_STATUS_CODES.has(err.code);
+    };
+
+    const onBatchResponse = (err: ServiceError | null) => {
+      if (isRetryable(err)) {
+        const backOffSettings =
+          options.gaxOptions?.retry?.backoffSettings ||
+          DEFAULT_BACKOFF_SETTINGS;
+        const nextDelay = getNextDelay(numRequestsMade, backOffSettings);
+        setTimeout(makeNextBatchRequest, nextDelay);
         return;
       }
 
-      if (mutationErrorsByEntryIndex.size !== 0) {
-        const mutationErrors = Array.from(mutationErrorsByEntryIndex.values());
-        err = new PartialFailureError(mutationErrors);
+      if (numRequestsMade !== 0) {
+        // If there's a race condition where all the mutations
+        // succeeded, but the server returned an error (like deadline
+        // exceeded), set error to null
+        if (pendingEntryIndices.size === 0 && err) {
+          err = null;
+        }
+
+        if (mutationErrorsByEntryIndex.size !== 0) {
+          const mutationErrors = Array.from(
+            mutationErrorsByEntryIndex.values()
+          );
+          callback(new PartialFailureError(mutationErrors));
```

Comment:
This loses the rpcError. I would try to add an extra attribute to PartialFailureError. Or just return the rpcError